### PR TITLE
BCON-10046 delete any previous comments and post a new comment

### PR DESCRIPTION
**Pull request by denivanov**

### CODE REVIEW

File: lib/HackTest.rb [hunk 1]
```diff
@@ -17,7 +19,13 @@ def my_handler(_event_type, _headers, data)
       data_hash[:repository][:full_name],
       data_hash[:pull_request][:number]
     ) # TODO: handle pagination
-    search_files(files)
+    non_inclusive_words_found = search_files(files)
+
+    if data_hash.key?(:pull_request) &&
+        (data_hash[:action] == 'opened' || data_hash[:action] == 'ready_for_review' || data_hash[:action] == 'reopened' || data_hash[:action] == 'edited')
```
Comment (cwagner) [inline lib/HackTest.rb]: /nit would be nice to extract the list of actions that we care about to an Array constant and then just check if `data_hash[:action]` is in that list

File: lib/HackTest.rb [hunk 2]
```diff
@@ -32,7 +40,7 @@ def search_files(files)
       file_name = file_hash[:filename]
       search_lines(non_inclusive_words_found, lines, file_name)
     end
-    non_inclusive_words_found
+    return non_inclusive_words_found
```
Comment (cwagner) [inline lib/HackTest.rb]: Any reason for adding this? It's my understanding that Ruby doesn't conventionally use the `return` keyword. http://ruby-for-beginners.rubymonstas.org/writing_methods/return_values.html

File: lib/HackTest.rb [hunk 3]
```diff
@@ -56,4 +64,29 @@ def add_to_hash(found_words, word, file_name)
...
+  def remove_pr_comments(pr_payload)
+    comments = github_client.issue_comments(
+        pr_payload[:repository][:full_name],
+        pr_payload[:pull_request][:number])
+
+    comments.each do |comment|
+      github_client.delete_comment(
+          pr_payload[:repository][:full_name],
+          comment.to_h[:id])
+    end
```
Comment (cwagner) [inline lib/HackTest.rb]: Won't this delete _all_ comments, not just the comments created by the bot? Is there a way to filter by author?

File: lib/HackTest.rb (hunk 1 above)
Comment (cwagner) [reply lib/HackTest.rb]: Also I don't think 'ready_for_review' is necessary here? If I'm understanding these events correctly, we'd have already received an 'opened' event when the draft PR was opened, changing it to 'ready_for_review' doesn't mean the content has changed, it's just moved out of draft status.

File: lib/HackTest.rb (hunk 2 above)
Comment (denivanov) [reply lib/HackTest.rb]: bookmarking this 😂

**Review by cwagner — APPROVED** [review]

**Pull request merged**